internal/encryption/keyprovider/aws_kms/provider_test.go

```diff
@@ -0,0 +1,160 @@
+package aws_kms
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/kms"
+	"github.com/aws/aws-sdk-go-v2/service/kms/types"
+	awsbase "github.com/hashicorp/aws-sdk-go-base/v2"
+)
+
+// skipCheck checks if the test should be skipped or not based on environment variables
+func skipCheck(t *testing.T) {
+	// check if TF_ACC and TF_KMS_TEST are unset
+	// if so, skip the test
+	if os.Getenv("TF_ACC") == "" && os.Getenv("TF_KMS_TEST") == "" {
+		t.Log("Skipping test because TF_ACC or TF_KMS_TEST is not set")
+		t.Skip()
+	}
+}
+
+const testKeyPrefix = "tf-acc-test-kms-key"
+const testAliasPrefix = "alias/my-key-alias"
+
+func TestKMSProvider_Simple(t *testing.T) {
+	skipCheck(t)
+	ctx := context.TODO()
+
+	keyName := fmt.Sprintf("%s-%x", testKeyPrefix, time.Now().Unix())
+	alias := fmt.Sprintf("%s-%x", testAliasPrefix, time.Now().Unix())
+
+	// Constructs a aws kms key provider config that accepts the alias as the key id
+	providerConfig := Config{
+		KMSKeyID: alias,
+		KeySpec:  "AES_256",
+	}
+
+	// Mimic the creation of the aws client here via providerConfig.asAWSBase() so that
+	// we create a key in the same way that it will be read
+	awsBaseConfig, err := providerConfig.asAWSBase()
+	if err != nil {
+		t.Fatalf("Error creating AWS config: %s", err)
+	}
+	_, awsConfig, awsDiags := awsbase.GetAwsConfig(ctx, awsBaseConfig)
+	if awsDiags.HasError() {
+		t.Fatalf("Error creating AWS config: %v", awsDiags)
+	}
+
+	kmsClient := kms.NewFromConfig(awsConfig)
+
+	// Create the key
+	keyId := createKMSKey(ctx, t, kmsClient, keyName, awsBaseConfig.Region)
+	defer scheduleKMSKeyDeletion(ctx, t, kms.NewFromConfig(awsConfig), keyId)
+
+	// Create an alias for the key
+	createAlias(ctx, t, kmsClient, keyId, &alias)
+	defer deleteAlias(ctx, t, kms.NewFromConfig(awsConfig), &alias)
+
+	// Now that we have the config, we can build the provider
+	provider, metaIn, err := providerConfig.Build()
+	if err != nil {
+		t.Fatalf("Error building provider: %s", err)
+	}
+
+	// Now we can test the provider
+	output, meta, err := provider.Provide(metaIn)
+	if err != nil {
+		t.Fatalf("Error providing keys: %s", err)
+	}
+
+	if len(output.EncryptionKey) == 0 {
+		t.Fatalf("No encryption key provided")
+	}
+
+	if len(output.DecryptionKey) == 0 {
```

Comment:
This behavior is actually not correct. Since you are not providing an input meta, the key provider does not need to provide a decryption key.

Reply:
That's true, though I don't think it hurts to provide the decryption key for testing?  I imagine we will refactor this pattern at some point.

Reply:
I would be cautious because it could lead to confusing error messages. The error message on incorrect/incomplete metadata should be that there is no decryption kry, not that the decryption failed.